tests/unit/test_cv_facts_tools.py

```diff
@@ -0,0 +1,157 @@
+        configlets_mappers=facts_unit.MOCKDATA_CONFIGLET_MAPPERS
+    )
+    yield database
+    LOGGER.debug('Final CVP state: %s', database)
+
+
+@pytest.fixture()
+def fact_unit_tools(request, cvp_database):
+    cvp_client = mock.get_cvp_client(cvp_database)
+    instance = CvFactsTools(cv_connection=cvp_client)
+    LOGGER.debug('Initial CVP state: %s', cvp_database)
+    yield instance
+    LOGGER.debug('Mock calls: %s', pprint.pformat(cvp_client.mock_calls))
+
+
+# ---------------------------------------------------------------------------- #
+#   TEST CASES
+# ---------------------------------------------------------------------------- #
+
+# CvFactsTools.__get_container_name
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+@pytest.mark.parametrize("container_def", generate_list_from_dict(facts_unit.MOCKDATA_CONTAINERS), ids=generate_test_ids_dict)
+def test_CvFactsTools__get_container_name(fact_unit_tools, container_def):
+    result = fact_unit_tools._CvFactsTools__get_container_name(key=container_def[mock.MockCVPDatabase.FIELD_KEY])
+    LOGGER.info('facts_tool response: %s', str(result))
+    assert result == container_def['name']
+    LOGGER.info('Container name for id %s is %s', str(container_def[mock.MockCVPDatabase.FIELD_KEY]), str(result))
+
+# CvFactsTools.__configletIds_to_configletName
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+@pytest.mark.parametrize("configlet_def", facts_unit.MOCKDATA_CONFIGLET_MAPPERS['data']['configlets'], ids=generate_test_ids_list)
+def test_CvFactsTools__configletIds_to_configletName(fact_unit_tools, configlet_def):
+    LOGGER.info('** Sending request to get name for configlet ID: %s', str(configlet_def[mock.MockCVPDatabase.FIELD_KEY]))
+    result = fact_unit_tools._CvFactsTools__configletIds_to_configletName(configletIds=[configlet_def[mock.MockCVPDatabase.FIELD_KEY]])
+    LOGGER.info('** facts_tool response: %s', str(result))
+    assert configlet_def['name'] in result
+    LOGGER.info('Configlet name for id %s is %s', str(configlet_def[mock.MockCVPDatabase.FIELD_KEY]), str(result))
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+def test_CvFactsTools__configletIds_to_configletName_empty(fact_unit_tools):
+    LOGGER.info('** Sending request to get name for configlet ID: None')
+    result = fact_unit_tools._CvFactsTools__configletIds_to_configletName(configletIds=[])
+    LOGGER.info('** facts_tool response: %s', str(result))
+    assert result == []
+    LOGGER.info('Configlet name for id [] is %s', str(result))
+
+# CvFactsTools.__device_get_configlets
+
+@pytest.mark.generic
+@pytest.mark.usefixtures("fact_unit_tools")
+@pytest.mark.usefixtures("cvp_database")
+@pytest.mark.parametrize("device_def", facts_unit.MOCKDATA_DEVICES, ids=generate_test_ids_list)
```

Comment:
`generate_test_ids_list` is identity, not sure it will work as intended. Maybe write a generic `get_test_id` for the facts_unit mock data that returns the correct key depending on the type (e.g. hostname for a device, name for a configlet...) ?